tsconfig.json

```diff
@@ -0,0 +1,70 @@
+    // "lib": [],                             /* Specify library files to be included in the compilation. */
+    // "allowJs": true,                       /* Allow javascript files to be compiled. */
+    // "checkJs": true,                       /* Report errors in .js files. */
+    // "jsx": "preserve",                     /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
+    // "declaration": true,                   /* Generates corresponding '.d.ts' file. */
+    // "declarationMap": true,                /* Generates a sourcemap for each corresponding '.d.ts' file. */
+    // "sourceMap": true,                     /* Generates corresponding '.map' file. */
+    // "outFile": "./",                       /* Concatenate and emit output to single file. */
+    // "outDir": "./",                        /* Redirect output structure to the directory. */
+    // "rootDir": "./",                       /* Specify the root directory of input files. Use to control the output directory structure with --outDir. */
+    // "composite": true,                     /* Enable project compilation */
+    // "tsBuildInfoFile": "./",               /* Specify file to store incremental compilation information */
+    // "removeComments": true,                /* Do not emit comments to output. */
+    // "noEmit": true,                        /* Do not emit outputs. */
+    // "importHelpers": true,                 /* Import emit helpers from 'tslib'. */
+    // "downlevelIteration": true,            /* Provide full support for iterables in 'for-of', spread, and destructuring when targeting 'ES5' or 'ES3'. */
+    // "isolatedModules": true,               /* Transpile each file as a separate module (similar to 'ts.transpileModule'). */
+
+    /* Strict Type-Checking Options */
+    "strict": true,                           /* Enable all strict type-checking options. */
+    // "noImplicitAny": true,                 /* Raise error on expressions and declarations with an implied 'any' type. */
+    // "strictNullChecks": true,              /* Enable strict null checks. */
+    // "strictFunctionTypes": true,           /* Enable strict checking of function types. */
+    // "strictBindCallApply": true,           /* Enable strict 'bind', 'call', and 'apply' methods on functions. */
+    // "strictPropertyInitialization": true,  /* Enable strict checking of property initialization in classes. */
+    // "noImplicitThis": true,                /* Raise error on 'this' expressions with an implied 'any' type. */
+    // "alwaysStrict": true,                  /* Parse in strict mode and emit "use strict" for each source file. */
+
+    /* Additional Checks */
+    // "noUnusedLocals": true,                /* Report errors on unused locals. */
+    // "noUnusedParameters": true,            /* Report errors on unused parameters. */
```

Comment:
In our current project we have this set to `false, and have a separate `tsconfig.no-unused.json` that we use for our lint npm script.

The reason for this is to allow unused variables while developing and testing things, while still allowing strict checks during commits.

(note, this is also how I set up the eslint rules in my current project where eslint is run as part of webpack, same with no-console)

Reply:
Yep, it can be annoying while developing. So you say it will break the build since it's part of webpack? If that is the case my concern is that devs can easily bypass the precommit linter, and someone else is going to have a surprise when building.

Reply:
Yeah, in some project setups (like CRA), it uses the eslint-loader, which fails the build on errors.

So what i've done is mark them as `warnings` (which I still all fix when done with my work).

Bypassing pre-commit hooks is bad behaviour on purpose, so I'd rather educate people to not do that, and "blame" them whenever they commit with errors, than to hinder all developers in their development speed.

Reply:
I'm fine with that. But since it involves things that are out of control of this repo should we keep this on by default and turn off in setups like you mentioned?

Reply:
I'm fine with that, let's document this use case somewhere at least then :D